### PR TITLE
:boom: Make TemporaryFileUpload.submission non-nullable

### PR DESCRIPTION
**Pull request by sergei-maertens**
Part of #3283

We've provided a transition period from Open Forms 2.7, which ensures that a submission is stored. Support for legacy uploads is now removed.

The celery background jobs should by now have deleted all the old unclaimed uploads, unless absurdly large retention values have been set.

**Checklist**

Check off the items that are completed or not relevant.

- Impact on features

  - [x] Checked copying a form
  - [x] Checked import/export of a form
  - [x] Config checks in the configuration overview admin page
  - [x] Problem detection in the admin email digest is handled

- Release management

  - [x] I have labelled the PR as "needs-backport" accordingly

- I have updated the translations assets (you do NOT need to provide translations)

  - [x] Ran `./bin/makemessages_js.sh`
  - [x] Ran `./bin/compilemessages_js.sh`

- Commit hygiene

  - [x] Commit messages refer to the relevant Github issue
  - [x] Commit messages explain the "why" of change, not the how

### CODE REVIEW

**Comment by sergei-maertens** [comment]
-> need to add an upgrade check script

File: docs/installation/upgrade-300.rst [hunk 1]
```diff
@@ -103,3 +103,15 @@ Removal of /api/v2/location/get-street-name-and-city endpoint
...
+
+Legacy temporary file uploads no longer supported
+=================================================
+
+Before Open Forms 2.7.0, temporary file uploads (as created by the file upload form
+component) would not be related to the submission they were uploaded in. We call these
+legacy temporary file uploads, and support for them has been removed in Open Forms 3.0.
+
+The setting ``TEMPORARY_UPLOADS_REMOVED_AFTER_DAYS`` controls how long file uploads are
+kept. Ensure that this many days have passed since the last legacy upload before
```
Comment (vaszig) [inline docs/installation/upgrade-300.rst:115]:
```suggestion
kept. Ensure that many days have passed since the last legacy upload before
```

**Review by vaszig — CHANGES_REQUESTED** [review]
I think there are some more pieces left like here:
https://github.com/open-formulieren/open-forms/blob/master/src/openforms/formio/components/vanilla.py#L387

File: docs/installation/upgrade-300.rst (hunk 1 above)
Comment (sergei-maertens) [reply docs/installation/upgrade-300.rst:115]: "this" refers to the value of the setting here, e.g. "Ensure that 5 days have passed since..."

**Comment by sergei-maertens** [comment]
> I think there are some more pieces left like here: https://github.com/open-formulieren/open-forms/blob/master/src/openforms/formio/components/vanilla.py#L387

is it not showing up in the diff for you? I've remove this check since I came across it when <kbd>CTRL</kbd> + <kbd>F</kbd> on 'legacy'.

**Review by vaszig — APPROVED** [review]

**Pull request merged**